opentracing/src/main/java/opentracing/TraceContextDecoder.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Map;
+
+/**
+ * Decodes a trace context from binary or text formats.
+ *
+ * @see TraceContextEncoder
+ */
+// See https://github.com/opentracing/opentracing.github.io/issues/21 about folding codec together
+public interface TraceContextDecoder {
```

Comment:
> (): @adriancole I just noticed that you are encoding/decoding to/from a
*single byte[] and a single Map<...>, not two like py and go. Is that
deliberate? If not, let's change it; if so, I'm curious about the semantic
divergence.

There are in fact two values. I think you are mistaking the type parameter
as being single value.

We could make it `List<>`, but yeah it is weird enough already.